### PR TITLE
New webhook for proxy

### PR DESCRIPTION
**Pull request by piyush-garg**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

File: pkg/reconciler/proxy/proxy.go [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+			Name:  "NO_PROXY",
+			Value: os.Getenv("NO_PROXY"),
+		}}
+
+		if after.Spec.Containers != nil {
+			for i, container := range after.Spec.Containers {
+				newEnvs := updateAndMergeEnv(container.Env, proxyEnv)
+				after.Spec.Containers[i].Env = newEnvs
+			}
+		}
```
Comment (vdemeester) [inline pkg/reconciler/proxy/proxy.go:286]: Should we only set them in case the value is not empty ? 🤔

**Review by vdemeester — APPROVED** [review]

**Review by pradeepitm12 — COMMENTED** [review]
/lgtm

**Pull request merged**